lib/util/specAlgos.ts

```diff
@@ -0,0 +1,108 @@
+import type {
+  IDurationRepresentation,
+  IDateTimeRepresentation, ITimeZoneRepresentation,
+} from './InternalRepresentations';
+import { toUTCDate } from './InternalRepresentations';
+
```

Comment:
Add documentation as to what this file does.

Reply:
+ Might be interesting to rename/ remove the used functions and where possible make more speedy.